recipe/meta.yaml

```diff
@@ -27,25 +27,30 @@ outputs:
       run_exports:
         - {{ pin_subpackage('openmpi', min_pin='x.x.x', max_pin='x') }}
       ignore_run_exports_from:
-        - {{ compiler('cuda') }}
+        - {{ compiler('cuda') }}  # [linux64 and cuda_compiler_version != 'None']
     requirements:
       build:
         - {{ compiler('c') }}
         - {{ compiler('cxx') }}
         - {{ compiler('fortran') }}
-        - {{ compiler('cuda') }}  # [linux64]
+        - {{ compiler('cuda') }}  # [linux64 and cuda_compiler_version != 'None']
         - autoconf  # [unix]
         - automake  # [unix]
         - libtool   # [unix]
         - make  # [unix]
         - perl 5.26.2
+        - ucx 1.9.0  # [linux64]
+        - ucx-proc =*=gpu  # [linux64 and cuda_compiler_version != 'None']
+        - ucx-proc =*=cpu  # [linux64 and cuda_compiler_version == 'None']
```

Comment:
These should be host dependencies.

Reply:
Good catch, thanks Isuru! Will fix shortly.

Just curious, though, technically would there be any difference if ucx is listed as a build dependency? It currently does not have `run_export` so we're pinning it here explicitly.

Reply:
Yes, there is. `build` platform and `host` platform can be different when cross compiling and this works only because we are not cross compiling for linux64. Also, dependencies that `ucx` brings in might conflict with those in `host` because `-I$BUILD_PREFIX/include` is added which we never want to do.

Reply:
Thanks, I missed the part on $BUILD_PREFIX, good to know.